Makefile

```diff
@@ -211,6 +211,11 @@ check.docs:
 	@$(MAKE) docs
 	@tests/scripts/validate_modified_files.sh docs
 
+check.crds: crds
+	@tests/scripts/validate_modified_files.sh crd
+
+check.codegen: codegen
+	@tests/scripts/validate_modified_files.sh codegen
```

Comment:
I am inclined to agree with @travisn 's thoughts from huddle today here. While I generally agree that make targets that are valid locally and in CI are a good thing, this is an area where I don't see value due to how the targets work. 

To use `check.crds` as the example, what it does is run `make crds` and then check the git status. I don't see how this is better than just running `make crds`. Devs should be able to look at the git status themselves to determine if they need to check anything else in. I think ther are also ways in which this is a worse experience than just running `make crds`. It's likely that devs will often have many changes already staged, meaning that `check.crds` would often report failure even when a dev has already run `make crds`.

Personally, I believe we should not use `validate_modified_files.sh` locally for the reasons above.

If we want to focus on make target changes that will actually deliver benefits to developers -- especially newer developers who aren't as familiar with all the generating targets Rook has, I would suggest improving the generating targets themselves.

1. establish a pattern that all generating targets have an alias with `gen.xyz`
   - I've already noted that `codegen` and `crd` are common targets among controller projects, so we should keep those names, but we can add additional aliases for them like `gen.codegen`
2. create a top-level `make generate` target that runs all generators -- this will especially help newer contributors who might not know which specific generators to run
3. stretch goal: try to set up `make generate` such that `make codegen` and other targets run in parallel, to speed up the workflow -- codegen is the slowest generator, so running it at the same time as others process will save a little time

Reply:
@BlaineEXE, Thank you for sharing your thoughts in detail!

It seens to me that you are on the one hand suggesting to abandon this PR's specific effort and on the other hand to improve and systematise the generating targets.

Would you agree that the latter (improve the generating targets) would best be done in a separate PR instead of repurposing this one?

Reply:
A new PR is probably a good plan. We can start with a clean slate for discussions that way

Reply:
PR for systematic generate tatgets: #14922